packages/grid/x-data-grid-pro/package.json

```diff
@@ -38,7 +38,7 @@
   },
   "dependencies": {
     "@mui/utils": "^5.4.1",
-    "@mui/x-license-pro": "5.5.0",
+    "@mui/x-license-pro": "5.5.1",
```

Comment:
I feel like most projects with several libraries (`storybook`, `@apollo`, ...) consider that they can make breaking change of internal APIs across packages.
And therefore that all the packages should always have the same exact version.

For instance, with #3688, `@mui/x-data-grid-pro` have to use the exact version of `@mui/x-data-grid`. Otherwise, we would not be able to change internal apis in `@mui/x-data-grid` that are being used in `@mui/x-data-grid-pro`.

Reply:
I agree on that, it makes sense to use range here

Reply:
> And therefore that all the packages should always have the same exact version.

In MUI Core, we have been doing this during the alpha/beta phases. I think that it's a best practice that any project should follow during the non-stable phases (encouraged by Lerna).

For the stable phase that we are in now, considering that `@mui/x-license-pro` should only be used by us the point of Flavien is fair. Using a strict version allows us to not follow semver in `@mui/x-license-pro`, and simplify the mental overhead for us: we know that no older version of the data grid will use a future version of the license package. I'm resolving this proposal. The problem I have suggested that it could solve is probably not worth it.